src/Android/Xamarin.Android/MainActivity.cs

```diff
@@ -70,13 +71,42 @@ protected override void OnCreate(Bundle bundle)
                 // Set up the search filtering.
                 SearchView searchBox = FindViewById<SearchView>(Resource.Id.categorySearchView);
                 searchBox.QueryTextChange += SearchBoxOnQueryTextChange;
+
+                // Add a button that brings up settings.
+                Button settingsButton = FindViewById<Button>(Resource.Id.settingsButton);
+                settingsButton.Click += (s, e) => PromptForKey();
+
+                // Check the existing API key for validity.
+                CheckApiKey();
             }
             catch (Exception ex)
             {
                 Console.WriteLine(ex.Message);
             }
         }
 
+        private async void CheckApiKey()
```

Comment:
Consider making this an async Task and awaiting `CheckApiKey` in OnCreate(). Otherwise where are potential exceptions handled?